File: JavaScript/7js 3/modifiedSum.js
/*
|--------------------------------------------------------------------------
| Nth power rules them all!
|--------------------------------------------------------------------------
|
*/

function modifiedSum(a, n) {
   return a.map(i => i ** n).reduce((a, b) => a + b) - a.reduce((a, b) => a + b)
}


function modifiedSum(a, n) {
   return a.reduce((s, c) => s + Math.pow(c, n) - c, 0);
}


function modifiedSum(arr, n) {
   let res = 0;
   for (let item of arr) {
      res += item ** n - item;
   }
   return res;
}


function modifiedSum(a, n) {
   return a.reduce((acc, curr) => acc + curr ** n - curr, 0);
}


console.log(modifiedSum([1, 2, 3], 3), 30);
console.log(modifiedSum([1, 2], 5), 30);

File: Python/7py 2/modified_sum.py
"""
|--------------------------------------------------------------------------
| Nth power rules them all!
|--------------------------------------------------------------------------
|
"""

def modified_sum(a, n):
   return sum(list(map(lambda i: i ** n, a))) - sum(a)


def modified_sum(a, n):
   return sum([x ** n - x for x in a])


def modified_sum(a, n):
   return sum(i ** n for i in a) - sum(a)


def modified_sum(a, n):
   res = 0
   for x in a:
      res += x ** n - x
   return res


print(modified_sum([1, 2, 3], 3), 30)
print(modified_sum([1, 2], 5), 30)